### DISCUSSION TITLE
Advanced restart with distributed mesh

### DISCUSSION THREAD

**Discussion by simopier** [idaholab/moose#19021] [Q&A Meshing] [answered]
I saw on the MOOSE [documentation](https://mooseframework.inl.gov/application_usage/restart_recover.html) that the advanced restart option was only supported for serial meshes, but I was wondering if it could be possible to use advanced restart on a simulation that used a distributed mesh (with the same number of processors). 

I tried to do it, but I get the following error:
```
*** ERROR ***
ERROR: Neither one of the following files can be located:
        '<checkpoint_file_directory>/####_mesh.cpr/1/header.cpr' nor
        '<checkpoint_file_directory>/####_mesh.cpr'
If you are running a parallel job, double check that you've created a split for 1 ranks.
```
The reason is that because I am using a distributed mesh, the `####_mesh.cpr` directory contains a folder named with the number of splits, which is not equal to 1. The file `header.cpr` therefore exists, but in `16`, for example, not in `1`.

Is there a way to solve this issue and still use the advanced restart option, or if I should use the `Variable Initialization` method instead?

And as a follow up question, how do I make sure that the new simulation also uses a distributed mesh?

Thank you! Let me know if you need additional information.

**Comment by GiudGiud** [comment, marked as answer]
Hello

If it's not supported then we need to develop it. I'll try this soon just to check.
Can you please try the restart using distributed Exodus (nemesis output) in the meantime?

Guillaume

**Reply by GiudGiud** [reply]
Seems solutionUOs are not supported for distributed restart https://github.com/idaholab/moose/issues/3201
But distributed advanced restart (with checkpoint = recover) should be supported, we even have a recipe for it https://github.com/idaholab/moose/pull/9649
I think you can see it in action here: https://civet.inl.gov/job/842975/

**Reply by GiudGiud** [reply]
Any news on this?

**Reply by simopier** [reply]
Thanks for the insight @GiudGiud !

The error I mentioned above can be removed using:
```
parallel_type = DISTRIBUTED
skip_partitioning = true
```
in the `mesh` block. That should be enough to fix the error when SolutionUOs is not in use.

However, since I am using SolutionUOs to define a polycrystal I unfortunately cannot use the advance restart option, as you mentioned. When I try, I get an error with `EXIT CODE: 11` before calculating the first non-linear residual. 

So I will be using the `Variable Initialization` feature described [here](https://mooseframework.inl.gov/application_usage/restart_recover.html). 

Hopefully that helps anyone with a similar issue in the future.
